packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,190 @@
+// @flow strict
+import { type Node, Fragment, useRef, useState, useEffect, useId } from 'react';
+import Box from './Box.js';
+import TapArea from './TapArea.js';
+import Popover from './Popover.js';
+import Text from './Text.js';
+import Link from './Link.js';
+import Icon from './Icon.js';
+import Tooltip from './Tooltip.js';
+import Layer from './Layer.js';
+import TrapFocusBehavior from './behaviors/TrapFocusBehavior.js';
+import { type Indexable, CompositeZIndex, FixedZIndex } from './zIndex.js';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider.js';
+
+type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to describe the popover content. Used for [accessibility](https://gestalt.pinterest.systems/web/popover#ARIA-attributes) purposes.
+   */
+  accessibilityPopoverLabel: string,
+  /**
+   * Specifies the preferred position of the tooltip and popover relative to HelpButton. See [Popover's ideal direction variant](https://gestalt.pinterest.systems/web/popover#Ideal-direction) to learn more.
+   */
+  idealDirection?: 'up' | 'right' | 'down' | 'left',
+  /**
+   * If provided, displays a link at the bottom of the popover message.
+   * - `href` is the URL the hyperlink points to.
+   * - `text` is the displayed text for the link. See the [link variant](https://gestalt.pinterest.systems/web/helpbutton#With-a-link) for more details.
+   * - Optionally use `accessibilityLabel` to supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like "Click Here", or "Read More" can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates `aria-label`. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See [ Link's accessibility guidelines](https://gestalt.pinterest.systems/web/link#Accessibility) for more information.
+   * - Optionally provide an `onClick` callback, which is fired when the link is clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   */
+  link?: {|
+    accessibilityLabel?: string,
+    href: string,
+    onClick?: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+    text?: string,
+  |},
+  /**
+   * Informational content that's displayed when the user clicks on HelpButton.
+   */
+  text: string,
+  /**
+   * Specifies the z-index for HelpButton's tooltip and popover to resolve any layering issues with other elements. See the [zIndex variant](https://gestalt.pinterest.systems/web/helpbutton#With-Z-index) for more details.
+   */
+  zIndex?: Indexable,
+|};
+
+const DEFAULT_ZINDEX = 2;
```

Comment:
Why do we need `DEFAULT_ZINDEX = 2;`to be 2?

Reply:
We need handle the `Tooltip` and `Popover` overlays, so we control this based on zIndex. 
Without this number we can see this behaviour: 
![a-01](https://user-images.githubusercontent.com/16820917/217825867-3b8a73db-eafd-4a9c-b3cf-b73750bc8a7c.gif)

Reply:
is 2 because I'm reducing by 1 this number, and we don't accept negative ZIndex 🙏🏾